config/trees.yaml

```diff
@@ -351,232 +407,263 @@ build_configs:
 
-  linusw_devel:
+  linusw: &linusw
+    <<: *base-arm
     tree: linusw
+
+  linusw_devel:
+    <<: *linusw
     branch: 'devel'
 
   linusw_fixes:
-    tree: linusw
+    <<: *linusw
     branch: 'fixes'
 
   linusw_for-next:
-    tree: linusw
+    <<: *linusw
     branch: 'for-next'
 
+  linux-pci:
+    tree: linux-pci
+    branch: 'next'
+    architectures:
+      - x86_64
+      - arm64
+      - arm
+
   mainline:
+    <<: *base
     tree: mainline
     branch: 'master'
 
-  media_fixes:
-    tree: media
+  media-committers_fixes: &media-committers
+    <<: *base-standard
+    tree: media-committers
     branch: 'fixes'
 
-  media_next:
-    tree: media
+  media-committers_next:
+    <<: *media-committers
     branch: 'next'
 
-  media-committers_fixes:
-    tree: media-committers
+  media_fixes: &media
+    <<: *base-standard
+    tree: media
     branch: 'fixes'
 
-  media-committers_next:
-    tree: media-committers
+  media_next:
+    <<: *media
     branch: 'next'
 
   mediatek_for_next:
+    <<: *base-arm
     tree: mediatek
     branch: 'for-next'
 
   net-next:
+    <<: *base
     tree: net-next
     branch: 'main'
 
   netdev-testing:
+    <<: *base
     tree: netdev-testing
     branch: 'https://netdev.bots.linux.dev/static/nipa/branches-hw.json'
 
-  next_master:
+  next_master: &next
+    <<: *base
     tree: next
-    branch: 'master'
 
   next_pending-fixes:
-    tree: next
+    <<: *next
     branch: 'pending-fixes'
 
   omap:
+    <<: *base-standard
     tree: omap
     branch: 'for-next'
 
   padovan:
+    <<: *base-standard
     tree: padovan
     branch: 'for-kernelci'
 
   pm:
+    <<: *base-standard
     tree: pm
     branch: 'testing'
 
   qcom:
+    <<: *base-arm
     tree: qcom
     branch: 'for-next'
 
-  renesas:
+  renesas: &renesas
+    <<: *base-arm
     tree: renesas
-    branch: 'master'
 
   renesas_next:
-    tree: renesas
+    <<: *renesas
     branch: 'next'
 
-  riscv_fixes:
+  riscv_fixes: &riscv
     tree: riscv
     branch: 'fixes'
     architectures:
       - riscv
 
   riscv_for-next:
-    tree: riscv
+    <<: *riscv
     branch: 'for-next'
-    architectures:
-      - riscv
 
   robh:
+    <<: *base-standard
     tree: robh
     branch: 'for-kernelci'
 
   rppt:
+    <<: *base-standard
     tree: rppt
     branch: 'for-kernelci'
 
   sashal-next:
+    <<: *base
     tree: sashal-next
     branch: 'linus-next'
 
-  soc_fixes:
+  soc_fixes: &soc
+    <<: *base-arm
     tree: soc
     branch: 'arm/fixes'
 
   soc_for-next:
-    tree: soc
+    <<: *soc
     branch: 'for-next'
 
-  stable_5.4:
-    tree: stable
-    branch: 'linux-5.4.y'
-
-  stable_5.10:
-    tree: stable
-    branch: 'linux-5.10.y'
-
-  stable_5.15:
-    tree: stable
-    branch: 'linux-5.15.y'
-
-  stable_6.1:
-    tree: stable
-    branch: 'linux-6.1.y'
-
-  stable_6.6:
-    tree: stable
-    branch: 'linux-6.6.y'
-
-  stable_6.12:
-    tree: stable
-    branch: 'linux-6.12.y'
-
-  stable-rc_5.4: &stable-rc
+  stable-rc: &stable-rc
```

Comment:
If I understand correctly this entry was added to create a nice and clean base anchor for all build configurations of `stable-rc` tree even though previously there was no entry for `stable-rc:master`.

I agree this is a good approach for trees with dormant `master` branch (like `linux-cip` or `kernelci` above) but I'm not sure if this also applies to (semi-)active master branches.

Would it make sense to create a base anchor on a first build configuration entry using given tree? Let me know what you think (maybe this additional cost turns out negligible over time)

Reply:
Done.

Reply:
> > Would it make sense to create a base anchor on a first build configuration entry using given tree? Let me know what you think (maybe this additional cost turns out negligible over time)
> 
> You mean the cost of parsing this file?

Not at all - I thought of the additional builds triggered by those new "base anchors". I agree readability is paramount in this case of large YAMLs and might take priority over some extra builds.

Thanks for adjusting your changes and sorry for writing confusing comment, I should have proof-read it before posting :sweat_smile:

Reply:
> > You mean the cost of parsing this file?
> 
> Not at all - I thought of the additional builds triggered by those new "base anchors". I agree readability is paramount in this case of large YAMLs and might take priority over some extra builds.

@pawiecz, ahh, this makes more sense now.

I wasn't aware of this, or rather, I assumed (and it seems incorrectly) that these trees are simply configuration elements, and each has to be explicitly referenced in a job somewhere for it to be used.

But if these are picked verbatim, I must revisit adding the `_base` trees to facilitate reuse.

> Thanks for adjusting your changes and sorry for writing confusing comment, I should have proof-read it before posting 😅

No problem at all!